src/main/scala/beam/utils/UberOSMSpeedMapper.scala

```diff
@@ -0,0 +1,95 @@
+package beam.utils
+import beam.analysis.via.CSVWriter
+import com.typesafe.scalalogging.LazyLogging
+import org.matsim.api.core.v01.network.Network
+import org.matsim.core.network.NetworkUtils
+import org.matsim.core.network.io.MatsimNetworkReader
+import org.supercsv.io.CsvMapReader
+import org.supercsv.prefs.CsvPreference
+
+import scala.reflect.ClassTag
+
+object UberOSMSpeedMapper extends LazyLogging{
+
+  /*
+    We need to pass 4 argument
+    1. network file path
+    2. comma separate list of movement-segment file example
+    3. comma separate list of movement-speed files
+    4. output file name
+   */
+
+  def main(args: Array[String]): Unit = {
+
+    val networkFile = args(0)
+    val movementFiles = args(1)
+    val speedFiles = args(2)
+    val outputFile = args(3)
+
+    val network = NetworkUtils.createNetwork()
+    new MatsimNetworkReader(network).readFile(networkFile)
+    writeOsmMappingCsv(network, movementFiles , speedFiles , outputFile)
+  }
+
+  def writeOsmMappingCsv(network: Network , movementSegmentFile: String, speedFile: String, outputFile: String): Unit ={
+    val osmSegment = readOsmWayToSegmentFile(movementSegmentFile)
+    val osmSpeed = readOsmSpeed(speedFile)
+
+    val csvWriter = new CSVWriter(outputFile)
+    val writer = csvWriter.getBufferedWriter
+    writer.write("linkid,capacity,free_speed,length")
+
+    network.getLinks.forEach((linkId,link) => {
+      val attributes = link.getAttributes
+      val origid = attributes.getAttribute("origid")
+      if(origid != null) {
+        osmSegment.get(origid.toString).foreach(segment =>{
+          val speed = osmSpeed.get(segment)
+          speed.foreach(s => {
+            writer.newLine()
+            writer.write(s"${linkId.toString},,${s*0.44704},")
```

Comment:
Please let's add either some function for conversion or add some comment why this is here

Reply:
I have extracted this in separate variable and added comment as well for this